### PR TITLE
Remove unwanted `else` block from statefulset controller.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by neolit123 — APPROVED** [review]
/priority backlog
/lgtm

**Comment by neolit123** [comment]
@humblec 
there is an error related to the change:
> pkg/controller/controller_utils.go:166:21: undefined: err

File: pkg/controller/controller_utils.go [hunk 1]
```diff
@@ -162,9 +162,9 @@ type ControllerExpectations struct {
 func (r *ControllerExpectations) GetExpectations(controllerKey string) (*ControlleeExpectations, bool, error) {
 	if exp, exists, err := r.GetByKey(controllerKey); err == nil && exists {
 		return exp.(*ControlleeExpectations), true, nil
-	} else {
-		return nil, false, err
 	}
+	return nil, false, err
```
Comment (xichengliudui) [inline pkg/controller/controller_utils.go]: `err` cannot return externally.

**Comment by humblec** [comment]
@neolit123 @xichengliudui PTAL, I have removed the controller changes from this patchset. Thanks in advance!!

**Comment by SataQiu** [comment]
@humblec 
there is an error related to the change: 
`pkg/controller/statefulset/stateful_set_test.go:654:9: undefined: obj`

**Comment by humblec** [comment]
@SataQiu Done.. PTAL

**Comment by humblec** [comment]
@neolit123 Can you please approve this PR ?

**Comment by neolit123** [comment]
> @neolit123 Can you please approve this PR ?

the sig-app approvers can.
/assign kow3ns

**Pull request merged**